packages/components/src/SquiggleEditor.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React  from 'react';
+import * as ReactDOM  from 'react-dom';
+import { SquiggleChart } from './SquiggleChart'
+import { ReactCodeJar } from "react-codejar";
+import type { exportEnv } from '@quri/squiggle-lang'
+
+
+export interface SquiggleEditorProps {
+  /** The input string for squiggle */
+  initialSquiggleString? : string,
+  
+  /** If the output requires monte carlo sampling, the amount of samples */
+  sampleCount? : number,
+  /** The amount of points returned to draw the distribution */
+  outputXYPoints? : number,
+  kernelWidth? : number,
+  pointDistLength? : number,
+  /** If the result is a function, where the function starts */
+  diagramStart? : number,
+  /** If the result is a function, where the function ends */
+  diagramStop? : number,
+  /** If the result is a function, how many points along the function it samples */
+  diagramCount? : number,
+  /** The environment, other variables that were already declared */
+  environment?: exportEnv,
+  /** when the environment changes. Used again for notebook magic*/
+  onEnvChange?(env: exportEnv) : void
+}
+
+const highlight = (editor: HTMLInputElement) => {
+  let code = editor.textContent;
+  code = code.replace(/\((\w+?)(\b)/g, '(<font color="#8a2be2">$1</font>$2');
+  editor.innerHTML = code;
+};
+
+interface SquiggleEditorState {
+  expression: string,
+  env: exportEnv
+}
+
+export class SquiggleEditor extends React.Component<SquiggleEditorProps, SquiggleEditorState>{
+  constructor(props: SquiggleEditorProps) {
+    super(props)
+    let code = props.initialSquiggleString ? props.initialSquiggleString : ""
+    this.state = {expression: code, env: props.environment }
+
+  }
+  render() {
+    let {expression, env} = this.state
+    let props = this.props
+    return (
+     <div>
+      <ReactCodeJar 
+        code={expression} 
+        onUpdate={e => {
+          this.setState({expression: e})
+          }}
+        style={{
+            borderRadius: "6px",
+            width: "530px",
```

Comment:
I imagine the width, at very least, should be a prop.

Reply:
Also, are you sure you need to do all of this with ``style``, as opposed to css or React-css or similar?

Reply:
I probably should change to React-css/emotion, worth it for the consistency

Reply:
Now that I think of it, I'm mildly afraid that a framework like emotion won't work for our use case, because emotion creates a CSS file and adds it to the head, but observable requires running the code within a sandbox that can't access the outer DOM. I'll give it a few tries nevertheless.